### ISSUE TITLE
Re-Enable Map Battles

### ISSUE THREAD

**Issue by garrettluskey** [araex/BannerlordCoop#709]
# For New Contributors
[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

## Description

Allow AIs to battle along with player to join AI battles.

Eventually we will disable the ability for players to battle each other for a MVP release. (see #710)

## Intended Design

Re-enable and add syncing for `DisablePartyEncounterPatch`

## Location

Create a branch based from [development](https://github.com/Bannerlord-Coop-Team/BannerlordCoop)

## Related Issues

Blocks #710

## Requirements

N/A

## Additional information

N/A

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

**Comment by AndreasOhmer** [comment]
Triage check — issue is alive but partially superseded by the recent `1.3-interactions` merge.

**Still relevant:**
- References `development` (not a dead branch)
- MVP milestone due 2026-08-31 — still in the future
- @EgardA active assignee

**What's changed since this was filed (2023-12):**
- `DisablePartyEncounterPatch.cs` is still present and blocks **all** `StartPartyEncounter` calls unconditionally (just prints "Tried to start encounter" and returns false)
- After #1166 (`1.3-interactions`) merged, there's now a more nuanced disable layer in `source/GameInterface/Services/MapEvents/Patches/MapEventDisablePatches.cs` using `CallOriginalPolicy.IsOriginalAllowed()` — allows the server to authorize specific calls
- A full MapEvent-Sync infrastructure exists now: `MapEventSync.cs`, `MapEventRegistry.cs`, handlers, messages

So "re-enable and add syncing for DisablePartyEncounterPatch" might re-frame as: **remove the old all-blocking `DisablePartyEncounterPatch` and rely on the new policy-based gating**, plus wire MapEvent start through the server-authorized path.

Probably worth a quick maintainer review to confirm the new direction, then this becomes implementable. Won't take it on without game-testing capacity (no local MP setup) but happy to update the issue description if @EgardA confirms the new shape.

Sibling #710 stays blocked by this. #722 ("Player Interactions") looks much more obsolete — flagging separately.

New contributor, batch-triaging.
